reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md

```diff
@@ -0,0 +1,638 @@
+---
+description: This article describes how to create a feedback provider.
+ms.date: 07/18/2023
+title: How to create a feedback provider
+---
+# How to create a feedback provider
+
+PowerShell 7.4-preview.3 introduces the concept of feedback providers. A feedback provider is a
+PowerShell module that utilizes the `IFeedbackProvider` interface that will trigger on a user action
+and give feedback to the user.
+
+Feedback providers are PowerShell modules that can give suggestions to the user based on what they
+are trying to execute in the shell. They currently trigger when there is a success or failure of
+execution. Feedback providers can take the specific information from successes and failures and
+based on that information, trigger and provide feedback to the user.
+
+## System requirements
+
+To create and use a feedback provider, you must be using the following software versions:
+
+- PowerShell 7.4-preview.3 or higher - includes the feedback provider interface
+- .NET 8 SDK - 8.0.100-preview.3.23178.7 or higher
+
+## Overview of a feedback provider
+
+A feedback provider is a PowerShell binary module. The module must implement the
+`using System.Management.Automation.Subsystem.Feedback` interface. This interface declares the
+methods used to get feedback based on the command line input. Currently it can trigger on success
+and errors of execution and provide the necessary information for assisting the user. Feedback can
+be anything the module owner decides, it can be suggestions on better practices or remediation
+commands to fix the error. We have created a good blog post you can read,
+[What are Feedback Providers?](https://devblogs.microsoft.com/powershell/what-are-feedback-providers/),
+to learn more about what they are and some of the built in feedback providers. The architecture of a
+feedback provider is shown below:
+
+![Architecture](media/create-feedback-provider/feedbackproviderarch.png)
+
```

Comment:
@sdwheeler I am not sure how to do an "alert" of some kind to also highlight that the user needs to have the `PSFeedbackProvider` experimental feature enabled for this to work.

Reply:
Added